### ISSUE TITLE
build section is not optional

### ISSUE THREAD

**Issue by mparry** [mamba-org/boa#264]
I think the intention is that the `build` section in the v2 spec is optional, as it is under `conda-build`, but it looks like there's a bug in the implementation:

https://github.com/mamba-org/boa/blob/82c2fd7f2231d465151c1c45e80b7b23299d1758/boa/core/run_build.py#L212

This yields

```
TypeError: 'NoneType' object is not iterable
```

I think that it should be `ydoc.get("build", {})` instead.

**Comment by wolfv** [comment]
PR is welcome!

**Comment by wolfv** [comment]
Although we should make sure that default values are properly set (`build / number` shoudl be zero etc).

**Comment by mparry** [comment]
Yeah, I can create one - I was just trying to work through some other problems too (see #265).
